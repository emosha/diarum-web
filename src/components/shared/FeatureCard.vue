<template>
  <div class="feature-card">
    <embed :src="getIcon(icon)" />
    <div class="feature-card__text">
      {{text}}
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component

export default class FeatureCard extends Vue {
  @Prop() public text: string|undefined;
  @Prop() public icon: string|undefined;

  public getIcon(path: string) {
    return require(`@/assets/img/card-swiggle-with-${path}@0,25x.svg`);
  }
}
</script>

<style scoped lang="scss">
.feature-card {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background-color: white;
  border-radius: 10px;
  min-height: 300px;
  min-width: 300px;
  box-shadow: 0 0 24px rgba(91, 94, 222, 0.16);

  &__text {
    font-size: 1.2em;
    text-transform: uppercase;
    letter-spacing: 1.5px;
  }

  img {
    width: 100px;
    height: 100px;
  }
}
</style>
