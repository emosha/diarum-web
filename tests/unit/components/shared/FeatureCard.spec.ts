import { shallowMount } from '@vue/test-utils';
import FeatureCard from '@/components/shared/FeatureCard.vue';

describe('FeatureCard.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(FeatureCard, {
      propsData: {
        icon: 'smile',
        text: 'Text',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
