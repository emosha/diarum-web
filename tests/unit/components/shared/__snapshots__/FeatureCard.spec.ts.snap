// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureCard.vue renders correctly 1`] = `
<div class="feature-card"><embed src="[object Object]">
  <div class="feature-card__text">
    Text
  </div>
</div>
`;
